src/v/config/base_property.h

```diff
@@ -26,19 +26,26 @@ namespace config {
 
 class config_store;
 using required = ss::bool_class<struct required_tag>;
+using needs_restart = ss::bool_class<struct required_tag>;
 
 class base_property {
 public:
+    struct metadata {
+        required required{required::no};
+        needs_restart needs_restart{needs_restart::yes};
+    };
+
     base_property(
       config_store& conf,
       std::string_view name,
       std::string_view desc,
-      required req);
+      metadata meta);
 
     const std::string_view& name() const { return _name; }
     const std::string_view& desc() const { return _desc; }
 
-    const required is_required() const { return _required; }
+    const required is_required() const { return _meta.required; }
+    bool needs_restart() { return bool(_meta.needs_restart); }
```

Comment:
This one is bool because in the contexts where we use it, it always gets cast to bool immediately -- either going into a json result for schema, or being used with `|=` during status checks (`bool_class` doesn't have `|=`